Lee/create visual3

Hyukjoo-Lee opened the pull request:
[I've referenced this link](https://www.figma.com/file/TNqsfSN2F6twHtuodJCFsq/Inviteyou-(collaborate)?type=design&node-id=31-1540&mode=design&t=kAbRvf8yfNe7Unom-0)

- Designed a new visual3 template.
- Button functions associated with the form will be implemented later.
- Adjusted menu styles to eliminate hovering effects that occurred when users hovered over menu options
color change will occurs only when users actively click on a menu option

Hyukjoo-Lee committed:
fixed login function

Hyukjoo-Lee committed:
install react-toastify library

Hyukjoo-Lee committed:
just removed a comment and annotations.

Hyukjoo-Lee committed:
intermediate commit

Hyukjoo-Lee committed:
reserve conflict merge from main to lee/fix-login

Hyukjoo-Lee committed:
fixed login function

Hyukjoo-Lee committed:
fixed notification's close button function

Hyukjoo-Lee committed:
Update error message handling

Hyukjoo-Lee committed:
Created indication for which input is incorrect

Hyukjoo-Lee committed:
removed nested if-else statement

Hyukjoo-Lee committed:
created visual3 component and designed intro content

Hyukjoo-Lee committed:
designed calendar

Hyukjoo-Lee committed:
created contact page

Hyukjoo-Lee committed:
created application detail form page

Hyukjoo-Lee committed:
created gallery carousel

Hyukjoo-Lee committed:
Refined the container's margin, height, and text display

Hyukjoo-Lee committed:
created location section

Hyukjoo-Lee committed:
Enhanced responsiveness and font sizes, optimized content spacing, and improved image display in carousel

HanKyungSung commented:
The font size is not responsive. It works well on mobile view but not on the desktop view.

> Desktop view:
![image](https://github.com/HanKyungSung/inviteyou/assets/25420417/41b9fb06-069d-4882-ae0b-2619096fc125)

HanKyungSung commented:
Also, the top of the visual component is not responsive on the desktop view as well.
- Date box size
- Font for the text and alignment.

![image](https://github.com/HanKyungSung/inviteyou/assets/25420417/ee338997-3951-4edb-811c-3fb70c9f3b7d)

HanKyungSung commented on invitation-frontend/src/component/Visual3.tsx:
```diff
@@ -1,5 +1,723 @@
+
+  return (
+    <>
+      <MantineProvider
```
I am not sure if we need `MaintineProvider` here. If we want to use it, should we have the break points here as well?

Hyukjoo-Lee replied:
I referenced this component, ../invitation-frontend/src/component/SamHan.tsx and I realized that we utilized the MantineProvider to set specific global styles for the entire design of a template

However, if you are talking about responsive design here, setting global breakpoints would be a good idea.

HanKyungSung commented on invitation-frontend/src/component/Visual3.tsx:
```diff
@@ -1,5 +1,723 @@
+            className={classes.newVisual}
+            fluid
+            px={0}
+            size={RESPONSIVE_MOBILE ? 550 : 450}
```
This is related to the breakpoints in the `MantineProvider`. Should we make it responsive depending on the Mantines's breakpoint we define using provider? or should we use the constant value?

Hyukjoo-Lee replied:
I think there could be pros/cons

if we set Mantine's breakpoints, we can implement a consistent responsive design across all components.
But, they may not align perfectly with the specific design requirements.

HanKyungSung commented on invitation-frontend/src/component/Visual3.tsx:
```diff
@@ -1,5 +1,723 @@
+              </p>
+            </div>
+          </Container>
+          {/* TODO: Need to adjust text inside image */}
```
I see this TODO comment now. Do you think we can have those changes in this PR?

Hyukjoo-Lee replied:
As we discussed earlier, if arranging texts is difficult, we can place text in the middle. 
However, the issue was easily resolved with some style adjustments. 

I forgot to delete the comment before this PR. I'll delete it.

HanKyungSung commented on invitation-frontend/src/component/Visual3.tsx:
```diff
@@ -1,5 +1,723 @@
+          </Container>
+
+          <Container
+            style={{
```
I think we can refactor this inline style.

HanKyungSung commented on invitation-frontend/src/component/Visual3.tsx:
```diff
@@ -1,5 +1,723 @@
+              </tbody>
+            </table>
+          </Container>
+          <Container size={RESPONSIVE_MOBILE ? 550 : 450} py={85}>
```
I see we keep using the same-size props with the same value. Can we refactor them?

HanKyungSung commented on invitation-frontend/src/component/Visual3.tsx:
```diff
@@ -1,5 +1,723 @@
+                      height={80}
+                    />
+                  </figure>
+                  <ul style={{ marginLeft: 15 }}>
```
We may want to move away from the inline-style

HanKyungSung commented on invitation-frontend/src/component/Visual3.tsx:
```diff
@@ -1,5 +1,723 @@
+                    </li>
+                  </ul>
+                </div>
+                <ul style={{ display: 'flex' }}>
```
This is as well. I think we should create new styles for each section.
ex)`contact-info-section-styles`

HanKyungSung commented:
I didn't comment on every line that has an inline style. Can we refactor them as we discussed before?

Hyukjoo-Lee commented:
> The font size is not responsive. It works well on mobile view but not on the desktop view.
> 
> > Desktop view:
> > ![image](https://user-images.githubusercontent.com/25420417/263618109-41b9fb06-069d-4882-ae0b-2619096fc125.png)

Hyukjoo-Lee commented:
> Also, the top of the visual component is not responsive on the desktop view as well.
> 
> * Date box size
> * Font for the text and alignment.
> 
> ![image](https://user-images.githubusercontent.com/25420417/263618438-ee338997-3951-4edb-811c-3fb70c9f3b7d.png)



> > The font size is not responsive. It works well on mobile view but not on the desktop view.
> > > Desktop view:
> > > ![image](https://user-images.githubusercontent.com/25420417/263618109-41b9fb06-069d-4882-ae0b-2619096fc125.png)

Hyukjoo-Lee commented:
> > The font size is not responsive. It works well on mobile view but not on the desktop view.
> > > Desktop view:
> > > ![image](https://user-images.githubusercontent.com/25420417/263618109-41b9fb06-069d-4882-ae0b-2619096fc125.png)

- The current code is only checking whether the viewport width is at most 767 pixels defining a var 'const RESPONSIVE_MOBILE = useMediaQuery('(max-width: 767px)');'
Do you think we need to add extra media queries to deal with this problem?
e.g.,
```javascript
const RESPONSIVE_WIDTH = {
  mobile: '767px',
  tablet: '768px',  
  desktop: '1024px'
};
```

Hyukjoo-Lee committed:
intermediate commit

Hyukjoo-Lee committed:
modified font size for responsive design on the desktop view

Hyukjoo-Lee committed:
modified the top of the visual component responsive on the desktop view

Hyukjoo-Lee committed:
encapsulate the duplicated responsive properties to reduce redundancy

Hyukjoo-Lee committed:
Refactored inline styling

Hyukjoo-Lee committed:
Adjusted background image on the laptop view

Hyukjoo-Lee commented:
### Description

- Modified and enhanced the visual component to make it responsive on desktop view.
- Adjusted the positioning and sizing of various elements for better looking.
- Refactored inline styling as we discussed during the previous meeting.
- Encapsulate the duplicated responsive properties to reduce redundancy.

Hyukjoo-Lee committed:
intermediate commit

Hyukjoo-Lee committed:
Revert "intermediate commit"

This reverts commit 23d1525846f54585da120827d9d4a70dd4e28121.

Hyukjoo-Lee committed:
Modified theme param type and moved away optionStyle from Visual3 component
